exporter/exporter_test.go

```diff
@@ -421,3 +452,59 @@ func TestExporterStartNoMetricsSelected(t *testing.T) {
 		defer exp.Stop()
 	}
 }
+
+func testBasicAuth(t *testing.T, opts *NATSExporterOptions, testuser, testpass string, expectedRc int) error {
+	exp := NewExporter(opts)
+	if err := exp.Start(); err != nil {
+		return err
+	}
+	defer exp.Stop()
+
+	return checkExporterFull(t, testuser, testpass, exp.http.Addr().String(), false, expectedRc)
+}
+
+func TestExporterBasicAuth(t *testing.T) {
+	opts := getDefaultExporterTestOptions()
+	opts.ListenAddress = "localhost"
+	opts.ListenPort = 0
+	opts.GetVarz = true
+	opts.GetConnz = true
+	opts.GetSubz = true
+	opts.GetRoutez = true
+
+	s := pet.RunServer()
+	defer s.Shutdown()
+
+	// first try user/pass with no auth.
+	testBasicAuth(t, opts, "colin", "password", http.StatusOK)
+
+	// now try user/pass
+	opts.HTTPUser = "colin"
+	opts.HTTPPassword = "password"
+	if err := testBasicAuth(t, opts, "colin", "password", http.StatusOK); err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	// now failures...
+	if err := testBasicAuth(t, opts, "colin", "garbage", http.StatusUnauthorized); err != nil {
+		t.Fatalf("%v", err)
+	}
+	if err := testBasicAuth(t, opts, "garbage", "password", http.StatusUnauthorized); err != nil {
+		t.Fatalf("%v", err)
+	}
+	if err := testBasicAuth(t, opts, "", "password", http.StatusUnauthorized); err != nil {
+		t.Fatalf("%v", err)
+	}
+	if err := testBasicAuth(t, opts, "colin", "", http.StatusUnauthorized); err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	opts.HTTPPassword = "$2a$11$SITqoMbjeKK1y9iV6nhXa.fC8o/QXEqs7o7DMkFtVErp0aTn12o1y"
```

Comment:
Is that the cost 11? If so, way too high for Travis. You may get a timeout and cut off.

Reply:
Try 5 max.